packages/recs/src/defineStoreComponents.ts

```diff
@@ -0,0 +1,31 @@
+import { parseStoreConfig, StoreUserConfig } from "@latticexyz/cli/src/config/parseStoreConfig";
+import { TableId } from "@latticexyz/utils";
+import { ContractComponents } from "@latticexyz/std-client";
+import { World, Schema } from "./types";
+import { Type } from "./constants";
+import { defineComponent } from "./Component";
+
+// TODO: figure out how to also translate TS types
+export function defineStoreComponents(world: World, userConfig: StoreUserConfig): ContractComponents {
+  const config = parseStoreConfig(userConfig);
+
+  const components = Object.fromEntries(
+    Object.entries(config.tables).map(([tableName, table]) => {
+      // TODO: move this to a util to share this logic with cli/autogen
+      const tableId = new TableId(config.namespace, table.fileSelector);
+      // translate v2 schema type to v1 RECS generic type
+      const schema: Schema = Object.fromEntries(
+        Object.entries(table.schema).map(([fieldName, schemaType]) =>
+          // TODO: infer TS types from schema types, map to real key names?
+          [fieldName, Type.T]
```

Comment:
This relates to the question in `decodeDynamicField` - could we create a type map like v1 `ContractSchemaValueTypes` but mapping from `SchemaType` to recs's `Type` and use it to infer the recs type here, and also give the `components` object strong typing (probably just using `as` while infering the type directly from the config)?